x500/cert/acme/src/test/java/org/wildfly/security/x500/cert/acme/AcmeClientSpiTest.java

```diff
@@ -1343,10 +1370,10 @@ private ClientAndServer setupTestObtainCertificate() {
                 .addDirectoryResponseBody(DIRECTORY_RESPONSE_BODY)
                 .addNewNonceResponse(NEW_NONCE_RESPONSE)
                 .addNewAccountRequestAndResponse(QUERY_ACCT_REQUEST_BODY, QUERY_ACCT_RESPONSE_BODY, QUERY_ACCT_REPLAY_NONCE, ACCT_LOCATION, 200)
-                .orderCertificateRequestAndResponse(ORDER_CERT_REQUEST_BODY, ORDER_CERT_RESPONSE_BODY, ORDER_CERT_REPLAY_NONCE, ORDER_LOCATION, 201, false)
+                .orderCertificateRequestAndResponse(ORDER_CERT_REQUEST_BODY, ORDER_CERT_RESPONSE_BODY, ORDER_CERT_REPLAY_NONCE, missingLocationURL ? null : ORDER_LOCATION, 201, false)
                 .addAuthorizationResponseBody(AUTHZ_URL, AUTHZ_REQUEST_BODY, AUTHZ_RESPONSE_BODY, AUTHZ_REPLAY_NONCE)
                 .addChallengeRequestAndResponse(CHALLENGE_REQUEST_BODY, CHALLENGE_URL, CHALLENGE_RESPONSE_BODY, CHALLENGE_REPLAY_NONCE, CHALLENGE_LOCATION, CHALLENGE_LINK, 200, false, VERIFY_CHALLENGE_URL, CHALLENGE_FILE_CONTENTS, AUTHZ_URL, UPDATED_AUTHZ_RESPONSE_BODY, UPDATED_AUTHZ_REPLAY_NONCE)
-                .addFinalizeRequestAndResponse(FINALIZE_RESPONSE_BODY, FINALIZE_REPLAY_NONCE, FINALIZE_URL, FINALIZE_LOCATION, 200)
+                .addFinalizeRequestAndResponse(FINALIZE_RESPONSE_BODY, FINALIZE_REPLAY_NONCE, FINALIZE_URL, missingLocationURL ? null : FINALIZE_LOCATION, 200)
```

Comment:
I don't think the `FINALIZE_URL` gets set using the `getLocation` method. So I think this change can be removed.

Reply:
I ran it with the debugger and I think it is the other way around. If I remove the setting to null for the ORDER_LOCATION but set the FINALIZE_LOCATION to null, then it throws the exception. If I alternate these, no exception is thrown.

Reply:
Just checked, `getLocation` is indeed called when finalizing the order: https://github.com/wildfly-security/wildfly-elytron/blob/44144e5107698e270a5b5be94473056a320f9950/x500/cert/acme/src/main/java/org/wildfly/security/x500/cert/acme/AcmeClientSpi.java#L514-L515. So we can remove the change made to `orderCertificateRequestAndResponse` and keep the change to `addFinalizeRequestAndResponse`. Thanks, Sonia!

Reply:
Okay, sounds good! Thank you